hazelcast/test/src/compact_test.cpp

```diff
@@ -634,6 +650,351 @@ struct hz_serializer<compact::test::employee_dto> : public compact_serializer
 
 namespace compact {
 namespace test {
+struct CompactRabinFingerprintTest : public ::testing::Test
+{
+    template<typename T>
+    using entry_t = std::tuple<int64_t, T, uint64_t>;
+
+    static constexpr int     int32_min = std::numeric_limits<int>::min();
+    static constexpr int64_t int64_min = std::numeric_limits<int64_t>::min();
+
+    template<typename T>
+    void check_each(std::vector<entry_t<T>> entries)
+    {
+        for (const entry_t<T>& e : entries)
+        {
+            using namespace hazelcast::client::serialization::pimpl::rabin_finger_print;
+
+            auto fp_before = std::get<0>( e );
+            auto value     = std::get<1>( e );
+            auto expected  = std::get<2>( e );
+            auto fp_after  = fingerprint64(fp_before , value);
+
+            EXPECT_EQ(fp_after, expected);
+        }
+    }
+};
+
+TEST_F(CompactRabinFingerprintTest, test_i8_fingerprint)
+{
+    check_each(
+        std::vector<entry_t<byte>>
+        { // Before               Val   After(Expected)
+            {100                 , -5   , -6165936963810616235},
```

Comment:
may it be better to write the expected values as uint64_t?

Reply:
I change `uint64_t` to `int64_t`.

Fixed.